sympy/logic/boolalg.py

```diff
@@ -1637,6 +1807,188 @@ def POSform(variables, minterms, dontcares=None):
+    (i.e., the "Zhegalkin polynomial").
+
+    There are 2^n possible Zhegalkin monomials in n variables, since
+    each monomial is fully specified by the presence or absence of
+    each variable.
+
+    A given monomial's presence or absence in a polynomial corresponds
+    to that monomial's coefficient being 1 or 0 respectively.
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import anf_coeffs, monomial, Xor
+    >>> truthvalues = [0, 1, 1, 0, 0, 1, 0, 1]
+    >>> coeffs = anf_coeffs(truthvalues)
+    >>> coeffs
+    [0, 1, 1, 0, 0, 0, 1, 0]
+    >>> polynomial = Xor(*[monomial(k, ['a','b','c'])
+    ...         for k, coeff in enumerate(coeffs) if coeff==1])
+    >>> polynomial
+    Xor(b, c, And(a, b))
+
+    """
+
+    s = '{0:b}'.format(len(truthvalues))
+    n = len(s) - 1
+
+    if len(truthvalues) != 2**n:
+        raise ValueError('The number of truth values must be a power of two.')
+
+    coeffs = [[v] for v in truthvalues]
+
+    for i in range(n):
+        tmp = []
+        for j in range(2 ** (n-i-1)):
+            tmp.append(coeffs[2*j] +
+                list(map(lambda x, y: x^y, coeffs[2*j], coeffs[2*j+1])))
+        coeffs = tmp
+
+    return coeffs[0]
+
+
+def minterm(k, variables):
+    """
+    Return the k-th minterm.
+
+    Minterms are numbered by a binary encoding of the complementation
+    pattern of the variables. This convention assigns the value 1 to
+    the direct form and 0 to the complemented form.
+
+    Parameters
+    ==========
+
+    k : int or list of 1s and 0s (complementation pattern)
+    variables : list of variables
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import minterm
+    >>> from sympy.abc import x, y, z
+    >>> minterm([1, 0, 1], [x, y, z])
+    And(Not(y), x, z)
+    >>> minterm(6, ['x', 'y', 'z'])
+    And(Not(z), x, y)
+
+    """
+    if isinstance(k, int):
+        k = integer_to_term(k, len(variables))
+    variables = [sympify(v) for v in variables]
+    return _convert_to_varsSOP(k, variables)
+
+
+def maxterm(k, variables):
+    """
+    Return the k-th maxterm.
+
+    Each maxterm is assigned an index based on the opposite
+    conventional binary encoding used for minterms. The maxterm
+    convention assigns the value 0 to the direct form and 1 to
+    the complemented form.
+
+    Parameters
+    ==========
+
+    k : int or list of 1s and 0s (complementation pattern)
+    variables : list of variables
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import maxterm
+    >>> from sympy.abc import x, y, z
+    >>> maxterm([1, 0, 1], [x, y, z])
+    Or(Not(x), Not(z), y)
+    >>> maxterm(6, ['x', 'y', 'z'])
+    Or(Not(x), Not(y), z)
+
+    """
+    if isinstance(k, int):
+        k = integer_to_term(k, len(variables))
+    variables = [sympify(v) for v in variables]
+    return _convert_to_varsPOS(k, variables)
+
+
+def monomial(k, variables):
+    """
+    Return the k-th monomial.
```

Comment:
Each boolean function can be uniquely represented by a _Zhegalkin Polynomial_ (Algebraic Normal Form). 
The Zhegalkin Polynomial of the boolean function with _n_ variables can contain up to _2<sup>n</sup>_ monomials.
We can enumerate all the monomials. Each monomial is fully specified by the presence or absence of each variable.
For example, boolean function with four variables _(a, b, c, d)_ can contain up to 2<sup>4</sup>=16 monomials. The 13-th monomial is the product _abd_, because 13 in binary is _1, 1, 0, 1_.